### PR TITLE
Add lc_package columns to avoid timeout/deadlock

### PR DESCRIPTION
**Pull request by charsbar**
This PR is another fix for #406. Contrary to #424, which removes `LOWER()` so that PAUSE can use indices (but lose case-insensitivity), this PR adds indexed lc_package columns to keep the current case-insensitive search and let PAUSE use indices at the cost of some extra space.

We may need to replace some more for the web app, but I haven't thoroughly checked yet. Also, we may need to add a unique lc_pack_user index (or replace pack_user with lc_pack_user), but I'm still wondering which is better.

If you try this PR before merging, apply one-off-utils/schemachange-2024-04.sql.

### CODE REVIEW

File: lib/PAUSE.pm [hunk 1]
```diff
@@ -345,7 +345,7 @@ sub owner_of_module {
                    FROM mods where modid = ?},
                  primeur => qq{SELECT package,
                           userid
-                   FROM primeur where LOWER(package) = LOWER(?)},
+                   FROM primeur where lc_package = LOWER(?)},
```
Comment (wolfsage) [inline lib/PAUSE.pm:348]: Any reason you left LOWER(?) in this one place but changed all others to use `lc`?

**Review by wolfsage — APPROVED** [review]
I think this is fine. I have two thoughts:

* If we used a case insensitive colation for lc_package, we wouldn't have to remember to compare with `lc $var` each time, which maybe provides a little less risk
* If we want to test this more before deploy we can stand this up on a DO box and do some manual testing (if we haven't already)

File: lib/PAUSE.pm (hunk 1 above)
Comment (charsbar) [reply lib/PAUSE.pm:348]: Or it might be better to repeat selectrow_arrayref without the for loop.

Comment (wolfsage) [reply lib/PAUSE.pm:348]: Ah, I see. I think leaving LOWER(?) here is fine then

**Comment by andk** [comment]
Thanks, applied and deployed just now

**Pull request closed without merging**